src/.nuget/packageLoad.targets

```diff
@@ -0,0 +1,150 @@
+                                   NuGet.Config - configuration file
+                                   packages.config - old-facioned way to installed packages
+                                   project.json - new way to install packages
+          NuGetDir               - location to install packages
+          SolutionFile           - if set, pass to NuGet.exe to restore everything for the solution
+          SolutionRoot           - if SolutionFile is not set and this is set, process all project.json
+                                   files under this directory. Excludes project.json in NuGetConfigDir.
+          InstallationDependency - list of files that, when changed, need to cause reinstall
+          BuildToolsSemaphore    - Created when build tools are restored
+        1. Downloads NuGet.exe
+        2. Handle packages.config and download packages it specifies
+        3. Handle project.json installation
+        4. Handle the solution file if given
+        5. Handle project.json using dnu instead of NuGet.exe if the solution file is not provided
+  -->
+
+  <!-- These properties are related to NuGet -->
+  <PropertyGroup>
+    <NuGetToolPath Condition="'$(NuGetToolPath)'==''">$([System.IO.Path]::Combine($(NuGetDir), "NuGet.exe"))</NuGetToolPath>
+    <NuGetConfigFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"NuGet.Config"))</NuGetConfigFile>
+    <NuGetPackagesFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"packages.config"))</NuGetPackagesFile>
+    <NuGetProjectJsonFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"project.json"))</NuGetProjectJsonFile>
+    <NuGetProjectLockJsonFile>$([System.IO.Path]::Combine($(NuGetConfigDir),"project.lock.json"))</NuGetProjectLockJsonFile>
+    <NuGetCommonOptions>-ConfigFile &quot;$(NuGetConfigFile)&quot; -Verbosity detailed</NuGetCommonOptions>
+    <NuGetPackageInstallOptions>-OutputDirectory &quot;$(NuGetDir.TrimEnd('\').TrimEnd('\'))&quot;</NuGetPackageInstallOptions>
+    <NuGetRestoreCommand>$(NuGetToolPath) restore $(NuGetCommonOptions)</NuGetRestoreCommand>
+    <SolutionSemaphore>$(NuGetDir)Solution.semaphore</SolutionSemaphore>
+    <NugetDownloadURL>https://dist.nuget.org/win-x86-commandline/v3.2.0/nuget.exe</NugetDownloadURL>
+    <DownloadCommand Condition="'$(OsEnvironment)'=='Windows_NT'">powershell -noprofile -nologo -command "(new-object System.Net.WebClient).DownloadFile('$(NugetDownloadURL)', '$(NuGetToolPath)')"</DownloadCommand>
+    <DownloadCommand Condition="'$(OsEnvironment)'!='Windows_NT'">curl -sSL --create-dirs -o $(NuGetToolPath) $(NugetDownloadURL)</DownloadCommand>
+    <BuildToolsSemaphore>$(NuGetDir)BuildTools.semaphore</BuildToolsSemaphore>
+  </PropertyGroup>
+
+  <!-- These are properties for the dnu -->
+  <PropertyGroup>
+    <PackagesCache Condition="'$(OsEnvironment)'=='Windows_NT'">$([System.IO.Path]::Combine($(USERPROFILE), '.nuget', 'packages'))</PackagesCache>
```

Comment:
If I add Linux runtime to project.json files, nuget.exe on the solution fails. Running dnu succeeds. Since I did not add Linux runtime in the PR, this is not enabled and we currently don't use dnu on Windows, but it may (potentially) be enabled in the next PR unless I find how to use nuget.exe.